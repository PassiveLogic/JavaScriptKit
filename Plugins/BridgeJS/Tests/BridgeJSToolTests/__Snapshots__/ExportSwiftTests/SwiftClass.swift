// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.
@_extern(wasm, module: "bjs", name: "return_string")
private func _return_string(_ ptr: UnsafePointer<UInt8>?, _ len: Int32)
@_extern(wasm, module: "bjs", name: "init_memory")
private func _init_memory(_ sourceId: Int32, _ ptr: UnsafeMutablePointer<UInt8>?)

@_expose(wasm, "bjs_takeGreeter")
@_cdecl("bjs_takeGreeter")
public func _bjs_takeGreeter(greeter: UnsafeMutableRawPointer) -> Void {
    takeGreeter(greeter: Unmanaged<Greeter>.fromOpaque(greeter).takeUnretainedValue())
}

@_expose(wasm, "bjs_Greeter_init")
@_cdecl("bjs_Greeter_init")
public func _bjs_Greeter_init(nameBytes: Int32, nameLen: Int32) -> UnsafeMutableRawPointer {
    let name = String(unsafeUninitializedCapacity: Int(nameLen)) { b in
        _init_memory(nameBytes, b.baseAddress.unsafelyUnwrapped)
        return Int(nameLen)
    }
    let ret = Greeter(name: name)
    return Unmanaged.passRetained(ret).toOpaque()
}

@_expose(wasm, "bjs_Greeter_greet")
@_cdecl("bjs_Greeter_greet")
public func _bjs_Greeter_greet(_self: UnsafeMutableRawPointer) -> Void {
    var ret = Unmanaged<Greeter>.fromOpaque(_self).takeUnretainedValue().greet()
    return ret.withUTF8 { ptr in
        _return_string(ptr.baseAddress, Int32(ptr.count))
    }
}

@_expose(wasm, "bjs_Greeter_changeName")
@_cdecl("bjs_Greeter_changeName")
public func _bjs_Greeter_changeName(_self: UnsafeMutableRawPointer, nameBytes: Int32, nameLen: Int32) -> Void {
    let name = String(unsafeUninitializedCapacity: Int(nameLen)) { b in
        _init_memory(nameBytes, b.baseAddress.unsafelyUnwrapped)
        return Int(nameLen)
    }
    Unmanaged<Greeter>.fromOpaque(_self).takeUnretainedValue().changeName(name: name)
}

@_expose(wasm, "bjs_Greeter_deinit")
@_cdecl("bjs_Greeter_deinit")
public func _bjs_Greeter_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Greeter>.fromOpaque(pointer).release()
}