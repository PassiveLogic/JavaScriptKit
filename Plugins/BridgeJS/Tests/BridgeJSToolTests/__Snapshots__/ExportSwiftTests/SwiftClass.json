{
  "classes" : [
    {
      "constructor" : {
        "abiName" : "bjs_Greeter_init",
        "parameters" : [
          {
            "label" : "name",
            "name" : "name",
            "type" : {
              "string" : {

              }
            }
          }
        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_Greeter_greet",
          "name" : "greet",
          "parameters" : [

          ],
          "returnType" : {
            "string" : {

            }
          }
        },
        {
          "abiName" : "bjs_Greeter_changeName",
          "name" : "changeName",
          "parameters" : [
            {
              "label" : "name",
              "name" : "name",
              "type" : {
                "string" : {

                }
              }
            }
          ],
          "returnType" : {
            "void" : {

            }
          }
        }
      ],
      "name" : "Greeter"
    }
  ],
  "functions" : [
    {
      "abiName" : "bjs_takeGreeter",
      "name" : "takeGreeter",
      "parameters" : [
        {
          "label" : "greeter",
          "name" : "greeter",
          "type" : {
            "swiftHeapObject" : {
              "_0" : "Greeter"
            }
          }
        }
      ],
      "returnType" : {
        "void" : {

        }
      }
    }
  ]
}